src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -1222,6 +1223,18 @@ protected virtual void AnalyzeTarget(TContext context, IEnumerable<Skimmer<TCont
 
         public static void AnalyzeTargetHelper(TContext context, IEnumerable<Skimmer<TContext>> skimmers, ISet<string> disabledSkimmers)
         {
+            // This might the public API invoked by other tools
+            Uri uri = context.CurrentTarget.Uri;
+            string originalStr = uri.OriginalString;
+            string filePath = uri.GetFilePath();
+            var URLPercentEncodingRegex = new Regex(@"%[0-9][0-9A-Fa-f]", RegexOptions.Compiled | RegexOptions.CultureInvariant);
```

Comment:
>URLPercentEncodingRegex [](http://example.com/codeflow?start=16&length=23)

you should cache this regex. why create it on every call?

you shouldn't be using .net here for this pattern matching, please use the context regex engine.

Reply:
Good suggestion. Update the regex.